babel.config.js

```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  plugins: ["css-modules-transform"],
```

Comment:
This doesn't work as you might think. It makes those `.css` files parseable but the output looks smth like this:
```ts
var styles$3 = {
  "Backdrop": "ContextMenu-module__Backdrop___GcXEc",
  "ContextMenu": "ContextMenu-module__ContextMenu___2DJfT"
};
```

The final consumer has no way to actually load the `.css` files containing those class names (and the associated CSS declarations)

Reply:
Hmm. I'm still trying to get Parcel to build the website so I haven't noticed that yet.

Reply:
Okay got that sorted out.

Reply:
> The final consumer has no way to actually load the .css files containing those class names (and the associated CSS declarations)

Right, I see what you're talking about now.

What's the advised way to do CSS modules with Preconstruct? Is this not supported yet?

https://github.com/preconstruct/preconstruct/issues/417

Reply:
> What's the advised way to do CSS modules with Preconstruct? Is this not supported yet?

This. It doesn't account for this use case as the output becomes invalid without extra tooling on the consuming side of things. Take for example Remix - it only has added CSS modules support on Jan 4 this year (see the PR [here](https://github.com/remix-run/remix/pull/4852)). So even though it already **was** a bundler... this package wouldn't work with it up until that point.

A lot of bundler frameworks nowadays only bundle the app code in node but they try to leave `node_modules` to be resolved by node itself (as it should be capable of doing that, so what's the point of bundling node dependencies?). Some escape hatches are often allowed but at that point, we are losing the "just install this package and use it". 

But as I said - we might consider allowing this with an explicit opt-in flag... but at the same time, I have a hard time advising people to use it.

Reply:
I think I understand your POV and motivations here. Sounds like the Preconstruct recommendation is to just use a different tool to bundle/process CSS and ship it with its own entry point in the package?

Reply:
Nah, no worries. I really appreciate you sharing your perspective (and playing devil's advocate). I think I've come around to your POV after thinking about the added configuration step.

Reply:
Also thanks so much for the quick responses here! Didn't expect such a fast turnaround but I really appreciate it.